API.md

```diff
@@ -0,0 +1,66 @@
+# Octofet API
+
+## `class Octofet`
+
+Create an object of type `Octofet` to communicate with a particular [Octofet board](https://my.amperka.com/modules/octofet).
+
+### `Octofet(uint8_t pinCS, uint8_t devicesCount = 1, uint8_t* resetValues = NULL)`
+
+Constructs a new object of data bus Hardware SPI with arguments:
+
+- `pinCS`: pin controls to enabled and disabled device on bus. Any GPIO-pin can be used.
+- `devicesCount`: quantity of octets boards connected in one chain: logical output pin headers of one module to the inputs of another. If the `devicesCount` argument is omitted, it used only one board Octofet.
+- `resetValues`: states power switchers by default as 8-bit value. One bit for one power switch: 1 is turn-on and 0 is turn-off. If `resetValues` argument is adjusted, then when calling methods `begin()` or `reset()`, power switchers sets value by `resetValues`. If `resetValues` argument is omitted — power switchers sets is turn-off.
+
+### `Octofet(uint8_t pinCS, uint8_t pinMOSI, uint8_t pinSCK, uint8_t devicesCount = 1, uint8_t* resetValues = NULL)`
+
+Constructs a new object of data bus Software SPI with arguments:
+
+- `pinCS`: pin controls to enabled and disabled device on bus. Any GPIO-pin can be used.
+- `pinMOSI`: pin for sending data to the peripherals. Any GPIO-pin can be used.
+- `pinSCK`: pin the clock pulses which synchronize data transmission generated by the master. Any GPIO-pin can be used.
+- `devicesCount`: quantity of octets boards connected in one chain: logical output pin headers of one module to the inputs of another. If the `devicesCount` argument is omitted, it used only one board Octofet.
+- `resetValues`: states power switchers by default as 8-bit value. One bit for one power switch: 1 is turn-on and 0 is turn-off. If `resetValues` argument is adjusted, then when calling methods `begin()` or `reset()`, power switchers sets value by `resetValues`. If `resetValues` argument is omitted — power switchers sets is turn-off.
+
+Difference between Hardware and Software SPI is speed–with software SPI it will run much slower than hardware SPI, but Hardware SPI works only on specific pins: `MOSI`, `MISO`(not used in Octofet) and `SCK`. Use Software SPI only when Hardware SPI pins are used by other devices.
+
+### `void begin()`
+
+Initializes the given interface, prepares the board for communication. Important, call this method at least once, in your `setup()`.
+
+### `digitalWrite(uint8_t device, uint8_t pin, bool value)`
```

Comment:
If we make `device` the default argument and move it to the end of the list, we won’t need two `digitalWrite` functions.